src/components/forms/PlanForm/components/retirePlanForm.tsx

```diff
@@ -0,0 +1,215 @@
+    baseEntityId: planPayload.identifier,
+    details: {
+      planIdentifier: planPayload.identifier,
+    },
+    entityType: PLAN_DEFINITION,
+    eventDate,
+    eventType: RETIRE_PLAN,
+    formSubmissionId,
+    obs: [
+      {
+        fieldCode: RETIRE_REASON,
+        fieldDataType: 'text',
+        fieldType: FORM_SUBMISSION_FIELD,
+        formSubmissionField: RETIRE_REASON,
+        humanReadableValues: [],
+        parentCode: '',
+        saveObsAsArray: false,
+        set: [],
+        values: [retireReason],
+      },
+    ],
+    providerId,
+    type: EVENT_LABEL,
+    version: Date.now(),
+  };
+};
+
+/** Form initial values */
+const initialValues: RetiringFormValues = {
+  otherReason: '',
+  retireReason: '',
+};
+
+/** Retiring plan reasons Component */
+const RetirePlanForm = (props: RetirePlanFormProps) => {
+  const { payload, savePlan, cancelCallBack, setSubmittingCallBack } = props;
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      /* tslint:disable-next-line jsx-no-lambda */
+      onSubmit={(values, { setSubmitting }) => {
+        const service = new OpenSRPService(OPENSRP_EVENT_ENDPOINT);
+        const formPayload = generatePayload(values, payload);
+        service
+          .create(formPayload)
+          .then(() => {
+            setSubmitting(false);
+            cancelCallBack();
+            // we have not handled what will happen if saving of a plan fails
+            // and the event for retiring the plan is successfull
+            savePlan(payload, setSubmittingCallBack);
+          })
+          .catch((e: Error) => {
+            setSubmitting(false);
+            displayError(e, AN_ERROR_OCCURRED, false);
+          });
+        return formPayload;
+      }}
+    >
+      {({ errors, isSubmitting, values, isValid }) => (
+        <Form>
+          <FormGroup>
+            <Label for="retireReason">{RETIRE_PLAN_REASON}</Label>
+            <Field
+              required={true}
+              component="select"
+              name="retireReason"
+              id="retireReason"
+              className={errors.retireReason ? 'form-control is-invalid' : 'form-control'}
+            >
+              <option key="" value="">
+                ------------
+              </option>
+              {Object.entries(RetireReasonOptions).map(e => (
+                <option key={e[0]} value={e[0]}>
+                  {e[1]}
+                </option>
+              ))}
+            </Field>
+          </FormGroup>
+          {values.retireReason === otherReason && (
+            <FormGroup>
+              <Label for="otherReason">{OTHER_REASON_LABEL}</Label>
+              <Field
+                required={true}
+                name="otherReason"
+                id="otherReason"
+                className={errors.otherReason ? 'form-control is-invalid' : 'form-control'}
+              />
+            </FormGroup>
+          )}
+          <Row>
+            <Col md={6} />
```

Comment:
I think it may be weird to space it this way.  I would go for centered buttons that are a bit wider.  So maybe a grid that looks like:

`md-3, mod-3-button, md3-button, md-3`

Reply:
Addressed [here](https://github.com/onaio/reveal-frontend/pull/1493/commits/2e4a666e42e0b22141252ef8ecf5c802b94e40cc)